packages/babel-plugin-transform-es2015-template-literals/src/index.js

```diff
@@ -1,7 +1,28 @@
 export default function ({ types: t }) {
+
+  function buildConcatCallExressions(items) {
+    return items.reduce(function(left, right) {
```

Comment:
To avoid the `_withIdentifier`, can't we rewrite to:

```js
let first = true;
return items.reduce((left, right) => {
  // Use isLiteral instead, 3 and null, and etc.
  let canInsert = t.isLiteral(right);
  if (!canInsert && first) {
    canInsert = true;
    first = false;
  }
  if (canInsert) {
    left.arguments.push(right);
    return left;
  }
  return t.callExpression(
    t.memberExpression(left, t.identifier('concat')),
    [right]
  );
});
```